app/src/main/java/net/gsantner/opoc/util/ShareUtil.java

```diff
@@ -1132,7 +1134,9 @@ public void writeFile(final File file, final boolean isDirectory, final Callback
         try {
             FileOutputStream fileOutputStream = null;
             ParcelFileDescriptor pfd = null;
-            if (file.canWrite() || (!file.exists() && file.getParentFile().canWrite())) {
+            final boolean existingEmptyFile = file.canWrite() && file.length() < MIN_OVERWRITE_LENGTH;
+            final boolean nonExistingCreatableFile = !file.exists() && file.getParentFile().canWrite();
+            if (existingEmptyFile || nonExistingCreatableFile) {
```

Comment:
More explicit logic. Still needs more testing.

Reply:
@gsantner Please test this for all the scenarios you can think of. It seems to work for all that I have tried.